src/uneditable-section/section-item.tsx

```diff
@@ -1,17 +1,166 @@
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleClickableClick = () => {
+        if (maskLoadingState === "fail") {
+            if (onTryAgain) onTryAgain();
+        }
+
+        switch (displayMaskState) {
+            case "masked": {
+                if (onUnmask) {
+                    onUnmask();
+                }
+                setMaskState("unmasked");
+                break;
+            }
+            case "unmasked": {
+                if (onMask) {
+                    onMask();
+                }
+                setMaskState("masked");
+                break;
+            }
+        }
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const getMaskedValue = () => {
+        return StringHelper.maskValue(value, {
+            maskChar,
+            maskRange,
+            unmaskRange,
+            maskRegex,
+            maskTransformer,
+        });
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderMaskingState = () => {
+        switch (maskLoadingState) {
+            case "fail":
+                return (
+                    <>
+                        <ErrorIcon />
+                        <ErrorLabel>Error</ErrorLabel>
+                        <TryAgainLabel weight="semibold">
+                            Try again?
+                        </TryAgainLabel>
+                    </>
+                );
+            case "loading":
+                return (
+                    <>
+                        <Spinner />
+                        <LoadingLabel>Retrieving...</LoadingLabel>
+                    </>
+                );
+            default:
+                return (
+                    <>
+                        <Text.Body>
+                            {displayMaskState === "masked"
+                                ? getMaskedValue()
+                                : value}
+                        </Text.Body>
+                        <IconContainer>
+                            {displayMaskState === "masked" ? (
+                                <EyeIcon data-testid="masked-icon" />
+                            ) : (
+                                <EyeSlashIcon data-testid="unmasked-icon" />
+                            )}
+                        </IconContainer>
+                    </>
+                );
+        }
+    };
+
+    const renderContent = () => {
+        if (!displayMaskState) {
+            return <Text.Body>{value}</Text.Body>;
+        }
+
+        if (disableMaskUnmask) {
+            return <Text.Body>{getMaskedValue()}</Text.Body>;
```

Comment:
based on the description of the `displayMaskState` prop, would still need to render the unmasked value if the `displayMaskState === "unmasked"`